backend/graph/api/episodes.resolvers.go

```diff
@@ -199,31 +201,38 @@ func (r *episodeResolver) Files(ctx context.Context, obj *model.Episode, audioLa
 
 // Chapters is the resolver for the chapters field.
 func (r *episodeResolver) Chapters(ctx context.Context, obj *model.Episode) ([]*model.Chapter, error) {
-	i, err := r.Loaders.EpisodeLoader.Get(ctx, utils.AsInt(obj.ID))
-	if err != nil || !i.AssetID.Valid {
+	return r.getChapters(ctx, obj.ID)
+}
+
+// SkipToChapter is the resolver for the skipToChapter field.
+func (r *episodeResolver) SkipToChapter(ctx context.Context, obj *model.Episode) (*model.Chapter, error) {
+	chapters, err := r.getChapters(ctx, obj.ID)
+	if err != nil || obj == nil {
 		return nil, err
 	}
-	metadataItems, err := r.Loaders.TimedMetadataLoader.GetMany(ctx, i.TimedMetadataIDs)
-	if err != nil {
-		return nil, err
+	if len(chapters) == 0 {
+		return nil, nil
 	}
-	metadataItems = lo.Filter(metadataItems, func(i *common.TimedMetadata, _ int) bool {
-		return i.Type == "chapter"
-	})
-
-	r.Loaders.PhraseLoader.LoadMany(ctx, lo.Uniq(lo.Map(metadataItems, func(i *common.TimedMetadata, _ int) string {
-		return i.ContentType.Value
-	})))
 
-	var out []*model.Chapter
-	for _, tm := range metadataItems {
-		chapter, err := resolveChapter(ctx, r.Loaders, obj.ID, tm.ID)
-		if err != nil {
-			return nil, err
+	slices.SortStableFunc(chapters, func(i, j *model.Chapter) int {
+		return cmp.Compare(i.Start, j.Start)
+	})
+	var chapter *model.Chapter
+	for _, c := range chapters {
+		if c.Start > obj.Duration/2 {
+			break
+		}
+		if c.ContentType != nil && (c.ContentType.Code == common.ContentTypeTheme.Value || c.ContentType.Code == common.ContentTypeInterview.Value || c.ContentType.Code == common.ContentTypeSpeech.Value) {
```

Comment:
not sure what theme is. And not sure whether I should use time to understand what it means. Probably fine.

Reply:
the rest of the logic looks plausible.